modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -262,13 +262,20 @@ def preprocess_func(cls, func):
         """
         return ray.put(func)
 
-    def length(self):
+    def length(self, materialize=True):
```

Comment:
Let's make this change for `ip` method as part of a separate issue.

Reply:
In order to make this change it would be nice to have code that uses this.

Reply:
https://github.com/modin-project/modin/blob/0df5fbf7a66663110878633467e4488ad37b0337/modin/distributed/dataframe/pandas/partitions.py#LL96C34-L96C39

Reply:
It looks like we're working through cached values everywhere right now, you want to change that and use `ip(materialize=False)`?

By the way, can we return an empty dataframe here to speed up?
`self.apply(lambda df: pd.DataFrame([]))._ip_cache` instead of `self.apply(lambda df: df)._ip_cache`

Reply:
> It looks like we're working through cached values everywhere right now, you want to change that and use `ip(materialize=False)`?

Yes, I think we should use public api where possible instead of private.

> By the way, can we return an empty dataframe here to speed up?
self.apply(lambda df: pd.DataFrame([]))._ip_cache instead of self.apply(lambda df: df)._ip_cache

No, we have to know where the original partition lays on.

Reply:
> No, we have to know where the original partition lays on.

Why do you think that if we return the same object, then this will show what is needed, but if we return another object, then no?

Reply:
Ah, I see, we pass a partition to the remote function anyway. Then, it should work. I thought you just returned an empty df from the remote function without any argument.

Reply:
> 

I will create issue after the PR is merged.

Reply:
https://github.com/modin-project/modin/issues/5634